handlers/dimensions.go

```diff
@@ -247,36 +246,36 @@ func (f *Filter) DimensionSelector() http.HandlerFunc {
 			return
 		}
 
-		// count number of options in dataset API
-		opts, err := f.DatasetClient.GetOptions(ctx, userAccessToken, "", collectionID, datasetID, edition, version, name, 0, 1)
+		// count number of options for the dimension in dataset API
+		opts, err := f.DatasetClient.GetOptions(ctx, userAccessToken, "", collectionID, datasetID, edition, version, name, dataset.QueryParams{Offset: 0, Limit: 1})
 		if err != nil {
 			setStatusCode(req, w, err)
 			return
 		}
 
 		// if there are more than maxNumOptionsOnPage, then we need to use the hierarchy model
-		if isHierarchy && opts.TotalCount > maxNumOptionsOnPage {
+		if isHierarchy && opts.TotalCount > MaxNumOptionsOnPage {
 			f.Hierarchy().ServeHTTP(w, req)
 			return
 		}
 
-		selectedValues, err := f.GetDimensionOptionsFromFilterAPI(req.Context(), userAccessToken, collectionID, filterID, name)
+		dims, err := f.DatasetClient.GetVersionDimensions(req.Context(), userAccessToken, "", collectionID, datasetID, edition, version)
```

Comment:
Not part of this PR - but an easy performance win for page load times could be to make these API requests concurrent.

Reply:
good point! I can do it in this PR